src/ape/utils.py

```diff
@@ -460,6 +460,23 @@ class AbstractDataClassMeta(DataClassMeta, ABCMeta):
 errors will occur. This class cannot be instantiated on its own.
 """
 
+
+def raises_not_implemented(fn):
+    """
+    A decorator to use over API methods that you want to be callable
+    but that are not actually implemented.
+
+    An example is when calling abstracts methods from the
+    :class:`~ape.api.providers.TestProviderAPI` mixin but when using a
+    provider that is not for testing.
+    """
+
+    def inner(*args, **kwargs):
+        raise NotImplementedError(fn.__name__)
```

Comment:
```suggestion
        raise NotImplementedError(f"Attempted to call method '{fn.__name__}' in 'ProviderAPI', which is only available in 'TestProviderAPI'.")
```

Reply:
But then it's not really a util, is it? lol

Reply:
I just moved it back, I like the longer messaging